### DISCUSSION TITLE
Exporting private collections that other users have made

### DISCUSSION THREAD

**Discussion by routerino** [outline/outline#7805] [Self Hosting] [answered]
We're looking at using outline internally as a knowledgebase and noticed that exports, even exports with an admin ui, do not catch private collections that users haven't exposed to the admin team.

This is problematic as exports will be our primary way of handling data retention long term.

Is there a way to either:
* have a superadmin key that will catch everything, including other people's private collections
* configure outline to force private collections to add administrators as a group on creation

**Comment by tommoor** [comment]
I forget, are you self-hosting? It's posted in that category so I'm going to assume so – why don't you just take a backup of the entire database?

**Reply by routerino** [reply]
Mainly due to a difference between retention and recovery.

Scenario: user wants us to restore an article that got deleted 2 months ago in his private collection. With regular exports that's relatively easy. With dB backups, it's not really gonna happen.

Second scenario: person leaves and whoever replaces him (after a month) wants access to his articles, but the private collection with all the documentation never got shared. What now?

Third: we want to rebuild outline in a different configuration and migrate the data. We export then import, then get a dozen complaints that a bunch of private collections we weren't aware of have disappeared.

**Reply by tommoor** [reply, marked as answer]
> Scenario: user wants us to restore an article that got deleted 2 months ago in his private collection

One thing we should do here is make the permanent deletion timeline configurable, currently it's hardcoded at 30 days.

> Second scenario: person leaves and whoever replaces him (after a month) wants access to his articles

In the licensed edition admins have the ability to manage all collections, which solves this issue.

> Third: we want to rebuild outline in a different configuration and migrate the data

I still feel like cloning the database might be a better solution here, but it does depend on your usecase a lot. Unfortunately there are literally opposite competing requirements with this one as many customers explicitly do not want any access to private collections.
